mypy_baseline/commands/_sync.py

```diff
@@ -25,14 +25,18 @@ def run(self) -> int:
                 continue
             if self.config.is_ignored_category(error.category):
                 continue
-            clean_line = error.get_clean_line(self.config)
+            clean_line = error.get_clean_line_components(self.config)
             new_baseline.append(clean_line)
 
+        if self.config.sort_baseline:
+            new_baseline.sort()
+        new_baseline_lines = [components.render() for components in new_baseline]
```

Comment:
Wouldn't sorting the generated text line work the same? Because it looks like the order of components in the generated line is the same as you use for sorting.

```suggestion
            clean_line = error.get_clean_line(self.config)
            new_baseline.append(clean_line)

        if self.config.sort_baseline:
            new_baseline.sort()
```

Reply:
I used the components rather than the generated text line to get numeric ordering for line numbers, e.g.

```
foo.py:1: an error
foo.py:2: bar
foo.py:10: qux
```

Obviously this only affects those who enable `--preserve-position`, and they'll probably have pretty noisy diffs anyway unless they never move their code around ;-P

Reply:
If the end goal is to reduce git conflicts then the only thing that matters is that the sorting is stable. It's ok if it's a bit off.

Reply:
Fair enough - have stripped it back to just sorting by text

Reply:
Beautiful!